linked-lists/c-or-cpp/Insert_and_delete_beginning.c

```diff
@@ -0,0 +1,65 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+//  Structure Definition
+
+// 1. Data of Integer type.
+// 2. Pointer to the Next Node .
+struct node {
+    int data ;
+    struct node *next ;
+} ;
+struct node *head = NULL ;
+struct node *current = NULL ;
+
+//Function to Print the Linked List
+void printList(){
+    printf("\n Item in the Linked List are :  ");
+    // Intialize the ptr(pointer) with head location .
+    struct node *ptr = head ;
+    // While Loop until we encounter a Node which is NULL .
+    // NULL ,  signifies we are the end of the list .
+    while(ptr!=NULL){
+        printf("%d ",ptr->data) ;  //Print the value of the Node
+        ptr = ptr->next ;         //Increment the ptr
+    }
+}
+
+//Insertion at the Beginning of the Linked List
+
+void insertBeg(int data){
+ //Creating a link
+ //Malloc is for space allocation .
+ struct node *link = (struct node*) malloc(sizeof(struct node)) ;
+ link->data = data ;
+ //Point the link's pointer to the current head
+ link->next = head ;
+ //Update the Head to the node we want to insert at the beggining
+ head = link ;
+}
+
+//Deletion at the Beginning of the Linked List
+
+struct node* deleteBeg(){
```

Comment:
check for underflowing edge case for the list. ( when the delete operation will be called on empty list )
 
your program in the current state will let thee program delete the memory it haven't even allocated if the `deleteBeg()` is called more times than the elements in the list, which will lead to a segmentation fault in most cases ( undefined behaviour in others ).